.github/workflows/lint.yml

```diff
@@ -0,0 +1,30 @@
+name: Lint
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run Black
+        uses: psf/black@stable
+        with:
+          options: "--check --verbose"
+          src: "./src"
+          version: 24.8.0
+      - name: Setup Python 
+        uses: actions/setup-python@v3

```

Comment:
It looks like setup-python is on v5 right now. Should we switch to that? Going through the changelog, I don't see any changes that should break anything. This workflow is getting a deprecation warning due to nodejs version which is fixed in one of the recent setup-python versions.

Reply:
Here's the warning:
> The following actions use a deprecated Node.js version and will be forced to run on node20: actions/setup-python@v3. For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

Not sure if this will actually cause any issues, but it's probably better to err on the side of caution.

Reply:
Yep I agree, I had no idea v5 was out. Good catch